namer/marathon/src/main/scala/io/buoyant/namer/marathon/MarathonInitializer.scala

```diff
@@ -53,6 +62,15 @@ case class MarathonConfig(
 
   private[this] def getDst = dst.getOrElse(s"/$$/inet/$getHost/$getPort")
 
+  private[this] val secretKey = "DCOS_SERVICE_ACCOUNT_CREDENTIAL"
+  private[this] def getAuth =
+    sys.env.get(secretKey).flatMap { secret =>
+      Api.readJson[MarathonSecret](Buf.Utf8(secret)).toOption
+    }.collect {
+      case MarathonSecret(Some(loginEndpoint), Some(privateKey), Some("RS256"), Some(uid)) =>
```

Comment:
where does this magic string come from?  Should it be configurable?  At the least, I think it should be consted somewhere with a comment.